src/H5Cimage.c

```diff
@@ -2386,7 +2391,7 @@ H5C__reconstruct_cache_contents(H5F_t *f, H5C_t *cache_ptr)
 
     /* Decode metadata cache image header */
     p = (uint8_t *)cache_ptr->image_buffer;
-    if (H5C__decode_cache_image_header(f, cache_ptr, &p) < 0)
+    if (H5C__decode_cache_image_header(f, cache_ptr, &p, cache_ptr->image_len + 1) < 0)
```

Comment:
Why are you adding '+1' here and then '-1' in the code in  H5C__decode_cache_image_header?

I believe that you can remove both.

Reply:
The +1 is the real total size of the buffer, so it makes sense to pass it here as that's what the function should expect to receive. The -1 in the function is just because of how the `H5_IS_BUFFER_OVERFLOW` macro is implemented. They could both be removed, but I think that's making things a lot less clear and introducing the potential for trouble.

Reply:
The image_len is not the actual size of the image_buffer?

Reply:
I'm not sure if the extra byte allocated for the cache image buffer is necessary (see H5Cimage.c ~line 622), but if that +1 was removed then this becomes simple

Reply:
Agree - those '+1' in the allocations are incorrect.  Especially since we are reading, writing, and bcasting only 'image_len' everywhere in the code.